core/trino-main/src/main/java/io/trino/execution/QueryManager.java

```diff
@@ -36,6 +36,11 @@
     void addOutputInfoListener(QueryId queryId, Consumer<QueryExecution.QueryOutputInfo> listener)
             throws NoSuchElementException;
 
+    /**
+     * Notify that one of the output tasks failed for a given query
+     */
+    void outputTaskFailed(TaskId taskId, Throwable failure);
```

Comment:
Output task is a task that produces a final output for a query

Reply:
Is this a notification that the output task itself failed? Or does the output task fail if any of its source tasks fail (via a cascading propagation of the error), and therefore, the output task failing is a proxy for any task failing?

Reply:
It is a notification that the output task failed. 

Maybe let me elaborate more on why this is needed.

Generally speaking a task has 2 communication links:

- Task <-> Coordinator (for status updates)
- Task <-> Downstream Task (for sending the results)

Let's assume a scenario when a link between a task and a downstream task is broken (for whatever reason) while the link between a task and coordinator is not.

Without failure recovery the downstream task would discover that the communication link is broken and fail a query. 

However with failure recovery enabled the downstream task (task that is running on coordinator) is configured to ignore the failures and not fail the query. That may result into a "deadlock", when the coordinator thinks that a task if active, but since the communication link between the task and it's downstream task is broken nobody is pooling the results leaving it in a blocked state. Thus it is important to notify the scheduler about such communication failures so the scheduler can react and re-schedule a task.

Reply:
Added a comment at the place when the listeners are attached (`CoordinatorStagesScheduler#schedule`)